internal/objects/gateway/gateway.go

```diff
@@ -78,7 +79,7 @@ func ContourForGateway(ctx context.Context, cli client.Client, gw *gatewayv1alph
 		return nil, nil
 	}
 
-	cntr, err := objcontour.CurrentContour(ctx, cli, gc.Spec.ParametersRef.Namespace, gc.Spec.ParametersRef.Name)
+	cntr, err := objcontour.CurrentContour(ctx, cli, pointer.StringPtrDerefOr(gc.Spec.ParametersRef.Namespace, ""), gc.Spec.ParametersRef.Name)
```

Comment:
these were just quick changes based on the switch to pointers, so there might be some cleanup actually needed and new tests added

Reply:
Should probably be an error if there's no namespace in the `ParametersRef`, given that a `Contour` is a namespaced CRD.